File: app-miyou/src/main/java/com/miyou/app/infrastructure/inbound/web/credit/dto/ChargeByPaymentRequest.java
package com.miyou.app.infrastructure.inbound.web.credit.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

public record ChargeByPaymentRequest(
	@NotBlank String userId,
	@NotBlank String paymentKey,
	@NotBlank String orderId,
	@NotBlank String pgProvider,
	@Positive long amount
) {
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/inbound/web/credit/dto/ChargeByPaymentRequest.kt
package com.miyou.app.infrastructure.inbound.web.credit.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive

data class ChargeByPaymentRequest(
    val userId: String,
    val paymentKey: String,
    val orderId: String,
    val pgProvider: String,
    @field:Positive val amount: Long,
)
